integration/container/copy_linux_test.go

```diff
@@ -0,0 +1,71 @@
+package container // import "github.com/docker/docker/integration/container"
+
+import (
+	"archive/tar"
+	"bytes"
+	"context"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/integration/internal/container"
+	"github.com/docker/docker/pkg/mount"
+	"gotest.tools/assert"
+)
+
+// Makes sure copy does not leak shared mounts
+// Relates to https://github.com/moby/moby/issues/37624
+func TestCopyToContainerDoesNotLeakMounts(t *testing.T) {
```

Comment:
This test is only valid if `/` is a shared mount, which it is not in the integration testing environment, so I've been just manually running this test after making `/` in the test env shared.